test/ray_julia_jll/gcs_client.jl

```diff
@@ -1,44 +1,39 @@
 @testset "GCS client" begin
     using UUIDs
     using .ray_julia_jll: JuliaGcsClient, Connect, Put, Get, Keys, Exists, Status, ok,
-                          ToString
+                          ToString, Disconnect
 
     client = JuliaGcsClient("127.0.0.1:6379")
 
     ns = string("TESTING-", uuid4())
 
     # throws if not connected
-    @test_throws ErrorException Put(client, ns, "computer", "mistaek", false, -1)
-    @test_throws ErrorException Get(client, ns, "computer", -1)
-    @test_throws ErrorException Keys(client, ns, "", -1)
-    @test_throws ErrorException Exists(client, ns, "computer", -1)
+    @test_throws ErrorException Put(client, ns, "computer", "mistaek", false)
+    @test_throws ErrorException Get(client, ns, "computer")
+    @test_throws ErrorException Keys(client, ns, "")
+    @test_throws ErrorException Exists(client, ns, "computer")
 
     status = Connect(client)
     @test ok(status)
     @test ToString(status) == "OK"
 
-    @test Put(client, ns, "computer", "mistaek", false, -1) == 1
-    @test Get(client, ns, "computer", -1) == "mistaek"
-    @test Keys(client, ns, "", -1) == ["computer"]
-    @test Keys(client, ns, "comp", -1) == ["computer"]
-    @test Keys(client, ns, "comppp", -1) == []
-    @test Exists(client, ns, "computer", -1)
+    @test Put(client, ns, "computer", "mistaek", false) == 1
+    @test Get(client, ns, "computer") == "mistaek"
+    @test Keys(client, ns, "") == ["computer"]
+    @test Keys(client, ns, "comp") == ["computer"]
+    @test Keys(client, ns, "comppp") == []
+    @test Exists(client, ns, "computer")
 
     # no overwrite
-    @test Put(client, ns, "computer", "blah", false, -1) == 0
-    @test Get(client, ns, "computer", -1) == "mistaek"
+    @test Put(client, ns, "computer", "blah", false) == 0
+    @test Get(client, ns, "computer") == "mistaek"
 
     # overwrite ("added" only increments on new key I think)
-    @test Put(client, ns, "computer", "blah", true, -1) == 0
-    @test Get(client, ns, "computer", -1) == "blah"
+    @test Put(client, ns, "computer", "blah", true) == 0
+    @test Get(client, ns, "computer") == "blah"
 
     # throw on missing key
-    @test_throws ErrorException Get(client, ns, "none", -1)
+    @test_throws ErrorException Get(client, ns, "none")
 
-    # ideally we'd throw on connect but it returns OK......
-    badclient = JuliaGcsClient("127.0.0.1:6378")
-    status = Connect(badclient)
-
-    # ...but then throws when we try to do anything so at least there's that
-    @test_throws ErrorException Put(badclient, ns, "computer", "mistaek", false, -1)
```

Comment:
MWE
```julia
julia> using Ray.ray_julia_jll

julia> badclient = ray_julia_jll.JuliaGcsClient("127.0.0.1:6378")
Ray.ray_julia_jll.JuliaGcsClientAllocated(Ptr{Nothing} @0x00000001376270e0)

julia> ray_julia_jll.Connect(badclient)
[2023-10-20 18:04:32,822 E 28309 8652849] gcs_rpc_client.h:207: Failed to connect to GCS at address 127.0.0.1:6378 within 5 seconds.
OK

julia> [2023-10-20 18:05:27,877 E 28309 8653025] gcs_rpc_client.h:537: Failed to connect to GCS within 60 seconds. GCS may have been killed. It's either GCS is terminated by `ray stop` or is killed unexpectedly. If it is killed unexpectedly, see the log file gcs_server.out. https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure. The program will terminate.
(venv) MacBook-Air~/.j/d/Ray (gm/gcsclient|✔)

Reply:
The scenario in which this would occur is if you run `Ray.init()` without a local raylet present? If so, I'm fine with making this into an issue to tackle later.

Reply:
yeah and even then it should error earlier if it tried to get the GCS address during `Ray.init`